pkg/controller/kieapp/shared/utils.go

```diff
@@ -174,3 +180,52 @@ func GetNamespacedName(object metav1.Object) types.NamespacedName {
 		Namespace: object.GetNamespace(),
 	}
 }
+
+const (
+	TargetNamespace   = "openshift-console"
+	ConsoleYAMLSample = "ConsoleYAMLSample"
+)
+
+func GetConsoleYAMLSample(res resource.KubernetesResource) (*consolev1.ConsoleYAMLSample, error) {
+	kind := res.GetAnnotations()["consoleKind"]
+	if kind != ConsoleYAMLSample {
+		return nil, errors.New("Unrecognized Kind or Kind is missing.")
+	}
+
+	snippetStr := res.GetAnnotations()["consoleSnippet"]
+	var snippet bool = false
+	if snippetStr != "" {
+		tmp, err := strconv.ParseBool(snippetStr)
+		if err != nil {
+			return nil, errors.New("Unable to parse snippet to boolean.")
+		}
+		snippet = tmp
+	}
+	title, _ := res.GetAnnotations()["consoleTitle"]
+	desc, _ := res.GetAnnotations()["consoleDesc"]
+	name, _ := res.GetAnnotations()["consoleName"]
+
+	res.SetAnnotations(make(map[string]string))
```

Comment:
They may have their own annotations that need to be in the CR. Only remove our 3 known annotations.